### ISSUE TITLE
Open up public API to get SerializationService

### ISSUE THREAD

**Issue by iaptekar** [hazelcast/hazelcast#7919]
It would be nice to have a public way of getting the PortableContext from the Hazelcast configuration.

**Comment by jerrinot** [comment]
Hello @iaptekar,
can you please describe a use-case for this?

**Comment by iaptekar** [comment]
Yeah, we are currently creating some components that compare class definitions when a client connects to a cluster. This makes sure that the Portables we are using are correctly versioned. The only way we have discovered to find out what class definitions the cluster thinks it has is by casting the HazecastInstance to a HazelcastInstanceImpl and getting the SerializationService. We would like to not have to use casts to get to internal data we need.

**Comment by jerrinot** [comment]
I see. This will be tricky as it's considered the be an internal concern. Moving a class to public has consequences when it comes to compatibility guarantees, etc. 

However I do understand your motivation behind this request. Could you try to specify a minimal set of methods (including return types) you would need to make your checks possible? The smaller the set is then higher the chance it will make it into a public API. Do not feel to be constrained by the current internal API - feel free to design a new API where you see a fit.

**Comment by iaptekar** [comment]
Sure, we are only interested in the lookup method to get the ClassDefinition given factory id, class id and version.

**Comment by jerrinot** [comment]
@iaptekar: `ClassDefinition` looks very broad. What method would you need specifically? 

@sancar, @tombujok: guys, what do you think about this?

**Comment by iaptekar** [comment]
The only one we are using so far is PortableContext.lookupClassDefinition(factoryId, classId, version);

**Comment by lukasherman** [comment]
Another use case is to enable on-demand deserialization. Assume that there is a large object graph stored once as value and manipulated using processors. It is quite efficient to deserialize just part of the graph to update specific subtrees only. Value on-demand deserialization keeps serialized `Data`/`byte[]` along with `SerializationService` to deserialize when needed. 
A very quick and easy fix would be to handle `SerializationServiceAware` inside  `com.hazelcast.internal.serialization.impl.DefaultSerializationServiceBuilder#registerSerializerHooks`.
https://github.com/hazelcast/hazelcast/blob/0bc538045ecae2f79c68ebc9b91c5b80d2747954/hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/DefaultSerializationServiceBuilder.java#L326

including
```
 if (valueinstanceof SerializationServiceAware) {
```
